pkg/asset/cluster/cluster.go

```diff
@@ -92,32 +83,51 @@ func (c *Cluster) Generate(parents asset.Parents) (err error) {
 		}
 	}
 
-	timer.StartTimer("Infrastructure")
+	for _, r := range tfPlugin.Resources() {
 
-	stateFile, err := terraform.Apply(tmpDir, installConfig.Config.Platform.Name(), extraArgs...)
-	if err != nil {
-		err = errors.Wrap(err, "failed to create cluster")
-		if stateFile == "" {
-			return err
+		// Copy the terraform.tfvars to a temp directory where the terraform
+		// will be invoked within.
+		tmpDir, err := ioutil.TempDir("", fmt.Sprintf("openshift-install-%s-", r))
+		if err != nil {
+			return errors.Wrap(err, "failed to create temp dir for terraform execution")
+		}
+		//defer os.RemoveAll(tmpDir)
+
+		extraArgs := []string{}
```

Comment:
This seems more of a style recommendation than actual fix. I simply moved this block of code. I'm happy to change it, however, this is handled the same way in other files (with the same name!) the same. I can change all of them, or, even better, can we come up with a style guide (or use an existing one) so that style issues aren't an issue in PR's?

Reply:
It is a bit more than just style, but I am fine leaving it as is.

The reason why `var extraArgs []string` is generally better is because it does not allocate the array if the slice ends up not being written to. In this case, it would be even better to allocate the array to the size needed since we know the size at this point (ie, `len(terraformVariables.Files())`).

Reply:
I've updated the syntax to your preferred usage. The same usage is scattered in various other files I've seen and we should probably change it in those as well.